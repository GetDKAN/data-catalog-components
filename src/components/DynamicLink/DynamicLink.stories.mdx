import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import DynamicLink from './index';

<Meta title="Dynamic Link" component={DynamicLink} />

# Dynamic Link
This component creates a link that will be a Reach Router component if internal and a basic `a` tag if external.

<Canvas>
    <Story name="External">
      <DynamicLink
        url="https://demo.getdkan.org"
        content="Demo site"
        cue={<span> (external link)</span>}
      />
    </Story>
    <Story name="Internal">
      <DynamicLink
        url="/?path=/story/dynamic-link--external"
        content="External Link"
      />
    </Story>
</Canvas>