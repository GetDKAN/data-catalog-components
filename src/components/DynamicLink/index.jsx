import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import validator from "validator";

const DynamicLink = ({ url, content, cue, className }) => {
  if (validator.isURL(url, { require_protocol: true })) {
    return (
      <a className={className} href={url}>
        {content}
        {cue && cue}
      </a>
    );
  }
  return (
    <Link className={className} to={url}>
      {content}
    </Link>
  );
};

DynamicLink.defaultProps = {
  cue: undefined,
  className: "",
};

DynamicLink.propTypes = {
  url: PropTypes.string.isRequired,
  content: PropTypes.node.isRequired,
  cue: PropTypes.node,
  className: PropTypes.string,
};

export default DynamicLink;
