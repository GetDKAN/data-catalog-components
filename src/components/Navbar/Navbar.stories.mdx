import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import Navbar from './index';

<Meta title="USWDS/Navbar" component={Navbar} />

# Navbar

<Canvas>
  <Story name="Basic">
    <Navbar
      items={[
      {
      "label": "Home",
      "url": "/",
      "target": "_top"
      },
      {
      "label": "Datasets",
      "url": "/search",
      "target": "_top"
      },
      {
      "label": "Publishers",
      "url": "/publishers",
      "target": "_top"
      },
      {
      "label": "About",
      "url": "/about",
      "target": "_top"
      },
      {
      "label": "API",
      "url": "/api",
      "target": "_top"
      }
      ]}
      />
  </Story>
</Canvas>
