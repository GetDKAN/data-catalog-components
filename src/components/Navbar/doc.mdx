---
name: Navbar
menu: Components
route: /components/navbar
---

import { Playground, Props } from 'docz'
import Navbar from './index'

# Navbar Docs

The Navbar is a horizontal rectangle which contains a primary navigation menu and a menu toggling widget.

## Basic usage
<Playground>
<div>

</div>
</Playground>

## Navbar Variants
<Playground>
<div>

</div>
</Playground>


## Properties
<Props of={Navbar} />
