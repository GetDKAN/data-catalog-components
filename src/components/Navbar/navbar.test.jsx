import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from './index';
import DynamicLink from '../DynamicLink/index';

const testItems = [
  <a className="usa-current" href="#linkOne" key="one">
    Simple link one
  </a>,
  <a href="#linkTwo" key="two">
    Simple link two
  </a>,
  ]

const nav = {
    "main": [
      {
        "label": "Home",
        "url": "/",
        "target": "_top"
      },
      {
        "label": "Datasets",
        "url": "/search",
        "target": "_top"
      },
      {
        "label": "Publishers",
        "url": "/publishers",
        "target": "_top"
      },
      {
        "label": "About",
        "url": "/about",
        "target": "_top"
      },
      {
        "label": "API",
        "url": "/api",
        "target": "_top"
      }
    ]
  };

describe('Should render a  Navigation bar.', () => {
  test('renders', () => {
    render(<Navbar navLinks={nav.main} items={testItems} />);
  });
});
