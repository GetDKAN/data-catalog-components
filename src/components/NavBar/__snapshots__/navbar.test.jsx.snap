// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  className=" container-fluid main-navigation base-styles sc-bdVaJa jDHIWM"
>
  <nav
    className="navbar navbar-expand-md navbar-dark"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="mr-auto nav"
      >
        <li
          className="nav-item"
        >
          <a
            href="https://google.com"
          >
            Google
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="https://civicactions.com/"
          >
            CivicActions
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="https://github.com/"
          >
            Github
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
